pkl-core/src/main/java/org/pkl/core/util/GlobResolver.java

```diff
@@ -45,26 +45,24 @@ public InvalidGlobPatternException(String message) {
     }
   }
 
-  public static final class ResolvedGlobElement {
-    private final String path;
-
-    private final URI uri;
-    private final boolean isDirectory;
-
+  public record ResolvedGlobElement(String path, URI uri, boolean isDirectory) {
     public ResolvedGlobElement(String path, URI uri, boolean isDirectory) {
       this.path = path;
       this.uri = uri.normalize();
       this.isDirectory = isDirectory;
     }
 
+    @Deprecated
     public String getPath() {
       return path;
     }
 
+    @Deprecated
     public URI getUri() {
       return uri;
     }
 
+    @Deprecated
     public boolean isDirectory() {
       return isDirectory;
     }
```

Comment:
The util package is not a public API, so no need to add deprecated methods.

```suggestion
  public record ResolvedGlobElement(String path, URI uri, boolean isDirectory) {
  }
```

Reply:
Oh, we definitely want that, my bad!